### DISCUSSION TITLE
Display errors and warnings where they happen

### DISCUSSION THREAD

**Discussion by miv391** [godotengine/godot-proposals#7514] [Editor]
The way how errors and warnings are displayed could be improved. Important messages are currently very easy to miss. I would guess that especially beginner programmers just ignore the error and warning messages as there are so much more to think than to click Debugger tab and then Errors tab. These messages should not require two clicks to see. 

### Example

I wrote a simple program. I didn't run it yet. There is an error about expecting intended block. The error seems to have nothing to do with my code. Maybe I just ignore all these messages.

![kuva](https://github.com/godotengine/godot-proposals/assets/49998025/050a4151-71e7-4151-baeb-0fb95a1f53c6)

Lets run the program. For some reason the timer didn't do anything. Well, at least the red error disappeared.

![kuva](https://github.com/godotengine/godot-proposals/assets/49998025/f8369053-9991-45f9-8fde-5b8acd296896)

### Proposal

If the warning were displayed below the line that caused the warning, the bug would be quickly fixed. Example UI mockup:

![kuva](https://github.com/godotengine/godot-proposals/assets/49998025/4c7212d5-f992-4540-b8b0-66bd0d8591af)

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/1005 (see my comment there).
